packages/mcp-server-supabase/src/server.ts

```diff
@@ -382,6 +391,146 @@ export function createSupabaseMcpServer(options: SupabaseMcpServerOptions) {
           return await executeSql(project_id, query);
         },
       }),
+      list_edge_functions: tool({
+        description: 'Lists all Edge Functions in a Supabase project.',
+        parameters: z.object({
+          project_id: z.string(),
+        }),
+        execute: async ({ project_id }) => {
+          const response = await managementApiClient.GET(
+            '/v1/projects/{ref}/functions',
+            {
+              params: {
+                path: {
+                  ref: project_id,
+                },
+              },
+            }
+          );
+
+          assertSuccess(response, 'Failed to fetch Edge Functions');
+
+          // Fetch files for each Edge Function
+          const edgeFunctions = await Promise.all(
+            response.data.map(async (edgeFunction) => {
+              const response = await managementApiClient.GET(
+                '/v1/projects/{ref}/functions/{function_slug}/body',
+                {
+                  params: {
+                    path: {
+                      ref: project_id,
+                      function_slug: edgeFunction.slug,
+                    },
+                  },
+                  parseAs: 'arrayBuffer',
+                }
+              );
+
+              assertSuccess(
+                response,
+                'Failed to fetch Edge Function eszip bundle'
+              );
+
+              const deploymentId = getDeploymentId(
+                project_id,
+                edgeFunction.id,
+                edgeFunction.version
+              );
+              const pathPrefix = getPathPrefix(deploymentId);
+
+              const files = await extractFiles(
+                new Uint8Array(response.data),
+                pathPrefix
+              );
+
+              // Strip away path prefix so that we don't confuse the LLM
+              // TODO: do we need to do this for import_map_path too?
+              const entrypoint_path = edgeFunction.entrypoint_path
+                ? fileURLToPath(edgeFunction.entrypoint_path).replace(
+                    pathPrefix,
+                    ''
+                  )
+                : undefined;
+
+              return {
+                ...edgeFunction,
+                entrypoint_path,
+                files: await Promise.all(
+                  files.map(async (file) => ({
+                    name: file.name,
+                    content: await file.text(),
+                  }))
+                ),
+              };
+            })
+          );
+
+          return edgeFunctions;
+        },
+      }),
+      deploy_edge_function: tool({
+        description: `Deploys an Edge Function to a Supabase project. If the function already exists, this will create a new version. Example:\n\n${edgeFunctionExample}`,
+        parameters: z.object({
+          project_id: z.string(),
+          name: z.string().describe('The name of the function'),
+          entrypoint_path: z
```

Comment:
can we also support the `import_map_path`? Can be defaulted to `deno.json`.

Reply:
Import map path now supported. Logic is:
1. Use `import_map_path` if specified
2. Otherwise fallback to `import_map_path` on existing function if it exists
3. Otherwise use a filename heuristic to guess the path (`deno.json` first, then `import_map.json`)
4. Otherwise leave undefined